synapseclient/models/submissionview.py

```diff
@@ -0,0 +1,923 @@
+                been stored in Synapse by using the `activity` attribute on the View
+                and calling the `store()` method on the View instance. Adding an
+                activity to a snapshot of a table is meant to capture the provenance of
+                the data at the time of the snapshot. Defaults to True.
+            associate_activity_to_new_version: If True the activity will be associated
+                with the new version of the table. If False the activity will not be
+                associated with the new version of the table. Defaults to True.
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            A `TableUpdateTransaction` object which includes the version number of the snapshot.
+
+        Example: Creating a snapshot of a view with an activity
+            Create a snapshot of a view and include the activity. The activity must have been stored in
+            Synapse by using the `activity` attribute on the SubmissionView and calling the `store()`
+            method on the SubmissionView instance. Adding an activity to a snapshot of a view is meant
+            to capture the provenance of the data at the time of the snapshot.
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import SubmissionView
+
+            syn = Synapse()
+            syn.login()
+
+            view = SubmissionView(id="syn4567")
+            snapshot = view.snapshot(
+                label="Q1 2025",
+                comment="Submissions reviewed in Lab A",
+                include_activity=True,
+                associate_activity_to_new_version=True
+            )
+            print(snapshot)
+            ```
+
+        Example: Creating a snapshot of a view without an activity
+            Create a snapshot of a view without including the activity. This is used in
+            cases where we do not have any Provenance to associate with the snapshot and
+            we do not want to persist any activity that may be present on the view to
+            the new version of the view.
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import SubmissionView
+
+            syn = Synapse()
+            syn.login()
+
+            view = SubmissionView(id="syn4567")
+            snapshot = view.snapshot(
+                label="Q1 2025",
+                comment="Submissions reviewed in Lab A",
+                include_activity=False,
+                associate_activity_to_new_version=False
+            )
+            print(snapshot)
+            ```
+        """
+        # Replaced at runtime
+        return TableUpdateTransaction(entity_id=None)
+
+
+@dataclass
+@async_to_sync
+class SubmissionView(
```

Comment:
Should this include the `AccessControllable` mixin? What about handling for default columns?